### PR TITLE
Add transpilation step for javascript #6611

### LINKED ISSUE #6611: Add transpilation step for javascript
Transpilation would allow us to write javascript in es6. As discussed in the slack group, this will need to be incremental, only transpiling certain files as we convert them to es6 syntax.

### PR DESCRIPTION
**Pull request by li-kai**
Fixes #6611 

**Outline of Solution**

Babel is a module that allows conversion of javascript written in es6 to be transpiled to es5. I set up babel to transpile any files with the extension `.es6` in the `src/main/webapp/js` folder to `.js` extension with es5 syntax. E.g. `src/main/webapp/js/test.es6` would generate `src/main/webapp/js/test.js`.

Scripts are contained in `package.json` so there is no need to check for windows compatibility.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -291,21 +301,8 @@ war.dependsOn enhancerRun
 
 // STATIC ANALYSIS TASKS
 
-def isWindows = System.getProperty("os.name").toLowerCase().contains("windows")
```
Comment (wkurniawan07) [inline build.gradle]: Revert this and the build should be green, I believe.

Comment (li-kai) [inline build.gradle]: Yep, thanks for that catch!

File: .travis.yml [hunk 2]
```diff
@@ -46,8 +51,7 @@ after_failure:
   - ./gradlew generateFailedCmd
 
 after_success:
-  - ./gradlew jacocoReport
-  - bash <(curl -s https://codecov.io/bash)
+  - ./gradlew jacocoTestReport
```
Comment (wkurniawan07) [inline .travis.yml]: Revert this.

File: build.gradle [hunk 3]
```diff
@@ -530,28 +522,21 @@ task downloadFailedXml {
 // COVERAGE TASKS
 
 jacoco {
-    toolVersion = "0.7.8"
```
Comment (wkurniawan07) [inline build.gradle]: Revert all the changes past this line.

**Pull request merged**